src/utils/oauth2_util.py

```diff
@@ -508,3 +501,15 @@ def decrypt_client_credentials(encrypted_client_credentials):
         "client_id": client_credentials_split[0],
         "client_secret": client_credentials_split[1],
     }
+
+
+def validate_password(password, encoded_password):
+    ioc = Injector(DiProviders)
```

Comment:
Why are you doing this here? Do you the implications or impact of doing this?

Why don't you simply inject it from here?
```
class Oauth2RepositoryImpl:
    @inject
    def __init__(self, engine: Engine):
```